test/CMakeLists.txt

```diff
@@ -20,9 +20,19 @@ enable_testing()
 
 if(MSVC)
   # VS needs -C config to run correct
-  add_custom_target(tests "${CMAKE_CTEST_COMMAND}" -C $<$<CONFIG:Debug>:Debug>$<$<CONFIG:Release>:Release> -V -T Test VERBATIM)
+  if (PCL_DISABLE_GPU_TESTS)
```

Comment:
What about 2 different ifs?
```cmake
set(PCL_CTEST_COMMAND "${CMAKE_CTEST_COMMMAND}")
mark_as_advanced(PCL_CTEST_COMMAND)
if (PCL_DISABLE_GPU_TESTS)
  set(PCL_CTEST_COMMAND "${CMAKE_CTEST_COMMMAND} -E gpu")
endif()
```
And then use this variable for the MSVC check

Reply:
I'm also ok with this. The only thing I'm not ok is the exponential grow of the expressions every time we have an `if` condition to check e.g., you're checking for `MSVC` and `PCL_DISABLE_GPU_TESTS` you now have 2^2 expressions. This is what I want to avoid.

Reply:
I think only cache variables can/need to be marked as advanced. In this snippets, PCL_CTEST_COMMAND is just a local var.

Reply:
> you now have 2^2 expressions

My approach is 2 + 2 expressions, but it's still the same :sob:

Reply:
So did I did something wrong with the latest approach?

Reply:
Nah. I was just kidding around